common/src/main/java/com/wynntils/wynn/handleditems/annotators/game/CraftedGearAnnotator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.handleditems.annotators.game;
+
+import com.wynntils.handlers.item.ItemAnnotation;
+import com.wynntils.handlers.item.ItemAnnotator;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.utils.CappedValue;
+import com.wynntils.wynn.handleditems.items.game.CraftedGearItem;
+import com.wynntils.wynn.item.parsers.WynnItemMatchers;
+import com.wynntils.wynn.objects.Powder;
+import com.wynntils.wynn.objects.profiles.item.GearIdentification;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+
+public final class CraftedGearAnnotator implements ItemAnnotator {
+    private static final Pattern CRAFTED_GEAR_PATTERN = Pattern.compile("^§3(.*)§b \\[100%\\]À$");
+
+    @Override
+    public ItemAnnotation getAnnotation(ItemStack itemStack) {
+        String name = ComponentUtils.getCoded(itemStack.getHoverName());
+        Matcher matcher = CRAFTED_GEAR_PATTERN.matcher(name);
+        if (!matcher.matches()) return null;
+
+        CappedValue durability = WynnItemMatchers.getDurability(itemStack);
+
+        List<GearIdentification> identifications = new ArrayList<>();
+        List<Powder> powders = List.of();
+
+        // Parse lore for identifications and powders
+        List<Component> lore = ComponentUtils.stripDuplicateBlank(itemStack.getTooltipLines(null, TooltipFlag.NORMAL));
+        if (lore.size() <= 1) {
+            // We should always have the item name as the first line, unless some other mod interacts badly...
+            return null;
+        }
+        lore.remove(0); // remove item name
+
+        for (Component loreLine : lore) {
+            // FIXME: This is partially shared with GearAnnotator
+            String unformattedLoreLine = WynnUtils.normalizeBadString(loreLine.getString());
+
+            // Look for Powder
+            if (unformattedLoreLine.contains("] Powder Slots")) {
+                powders = Powder.findPowders(unformattedLoreLine);
+                continue;
+            }
+
+            // Look for identifications
+            Matcher identificationMatcher = GearAnnotator.ITEM_IDENTIFICATION_PATTERN.matcher(unformattedLoreLine);
```

Comment:
crafted items are going to need their own regex for identifications, since the format is different from regular gear (in other words, this matcher will never succeed)

Reply:
I started with a more ambitious goal in mind than what the code currently does, to handle crafted gear like we handle normal gear. But I realized after a while that this is more work than I expected. I still want to revisit this later on, but I might need to have a clear use in mind first.

I'll add a comment that this needs to be fixed, but for the rest I'll leave it as it is.